ramips: add mapping additional uci-defaults from stock firmware

Spudz76 opened the pull request:
Commit 1: base functions
>    base-files: functions: Add three new functions
>    Add mtd_get_raw_ascii() system function;
>    Add ucidef_set_ssid() and ucidef_set_reg_domain() uci-defaults functions

---

Commit 2: ramips/mt7621 platform specific (using above functions)
>    ramips: add mapping additional uci-defaults from stock firmware
>    initial support for parsing extra information from stock firmware during
>    first boot (`factory_extras`)
>    
>    one board is supported but others can be added if there are useful
>    settings to parse from the stock `devinfo` or similar partition
>    
>    * Linksys EA7300-v1:
>      * wireless: use default_ssid for all wireless (as on label)
>      * wireless: use cert_region for country by default (vs unset)

Signed-off-by: Tony Butler <spudz76@gmail.com>

mpratt14 commented:
hi again,

looks good

is there an existing function like  `mtd_get_raw_ascii` anywhere?

Spudz76 commented:
I will look around although I'm sure I already did.  There were definitely ones with additional parsing like for mac addresses (which validates formatting / parses further / not "raw").  I would have also been surprised there were none already, before writing them, so I'm going with there were none.  But I will check again just to be sure.

Spudz76 commented:
I checked and there is no existing generic string access function.

f00b4r0 commented:
This looks good but if I'm not mistaken this is essentially duplicating `mtd_get_mac_ascii()` except for the last call. Maybe `mtd_get_mac_ascii()` could be made to use the new function, to reduce duplicated code?

Spudz76 committed:
base-files: functions: Add three new functions

Add mtd_get_raw_ascii() system function;
Add ucidef_set_ssid() and ucidef_set_reg_domain() uci-defaults functions

Signed-off-by: Tony Butler <spudz76@gmail.com>

Spudz76 committed:
ramips: add mapping additional uci-defaults from stock firmware

initial support for parsing extra information from stock firmware during
first boot (`factory_extras`)

one board is supported but others can be added if there are useful
settings to parse from the stock `devinfo` or similar partition

* Linksys EA7300-v1:
  * wireless: use default_ssid for all wireless (as on label)
  * wireless: use cert_region for country by default (vs unset)

Signed-off-by: Tony Butler <spudz76@gmail.com>

Spudz76 commented:
> This looks good but if I'm not mistaken this is essentially duplicating `mtd_get_mac_ascii()` except for the last call. Maybe `mtd_get_mac_ascii()` could be made to use the new function, to reduce duplicated code?
No, that complicates the error message indicating the actual function that was called.  And changing a tried and true function doesn't make sense just to save a couple lines.

> Also you'll want to split these changes into at least 2 commits: separating the base-files package changes from the ramips ones.
Done.
